tests/unit/test-class-wc-payment-gateway-wcpay-subscriptions.php

```diff
@@ -226,10 +227,10 @@ function ( $id ) use ( $mock_subscription ) {
 			->with( self::USER_ID )
 			->willReturn( self::CUSTOMER_ID );
 
-		$this->mock_api_client
-			->expects( $this->once() )
-			->method( 'create_and_confirm_intention' )
-			->with( $this->anything(), $this->anything(), self::PAYMENT_METHOD_ID, self::CUSTOMER_ID, $this->anything(), false, false, $this->anything(), $this->anything(), true )
+		$request = $this->mock_wcpay_request( Create_And_Confirm_Intention::class );
```

Comment:
I love how simple it was to replace this API client mock with a request class!

However, please notice that there is a check for the payment method and customer IDs, as well as some boolean flags. I think we need to carry them over.

Reply:
Setters for crucial data like customer id, payment method id and some booleans have been added to test

Reply:
AFAICT the most crucial parameter this test is checking for is `off_session => true` (the last boolean true) in the call, as well as making sure `setup_future_usage` and `save_payment_method_to_platform` are not set: the payment method should be set up and saved already in this scenario.

So I'd keep only the tests for those parameters, as well as maybe the customer and payment method IDs. The rest don't matter.

The `save_payment_method_to_platform` parameter is now in the WooPay request class, so ideally that would warrant a whole separate test for that part, but I think we're okay to skip it.